### PR TITLE
Added Hmine algorithm

### PR DESCRIPTION
**Pull request by fatihsen20**
### Code of Conduct
New feature.

### Description

In order to enrich the library and to compare the speed and memory costs of certain algorithms, I added the hmine algorithm to the library.
### Related issues or pull requests
None

### Pull Request Checklist

- [x] Added a note about the modification or contribution to the `./docs/sources/CHANGELOG.md` file (if applicable)
- [x] Added appropriate unit test functions in the `./mlxtend/*/tests` directories (if applicable)
- [x] Modify documentation in the corresponding Jupyter Notebook under `mlxtend/docs/sources/` (if applicable)
- [x] Ran `PYTHONPATH='.' pytest ./mlxtend -sv` and make sure that all unit tests pass (for small modifications, it might be sufficient to only run the specific test file, e.g., `PYTHONPATH='.' pytest ./mlxtend/classifier/tests/test_stacking_cv_classifier.py -sv`)
- [x] Checked for style issues by running `flake8 ./mlxtend`

### CODE REVIEW

File: docs/sources/CHANGELOG.md [hunk 1]
```diff
@@ -931,4 +931,11 @@ imput arrays via `transform` and `fit_transform`
 
 ### Version 0.1.1 (2014-08-13)
```
Comment (rasbt) [inline docs/sources/CHANGELOG.md:932]: I think you accidentally added this to the wrong version in the Changelog

**Review by rasbt — CHANGES_REQUESTED** [review]
Awesome, thanks so much for the contribution! 

H-Mine is guaranteed to find the same frequent itemsets as apriori and fp-growth, correct? Would it be possible to add a unit test to test for equivalence with the existing algos?

**Comment by rasbt** [comment]
PS: Don't worry about the linter above, we can address that (happy to take care of that) at the very end.

**Comment by rasbt** [comment]
Thanks a lot for the update! I will do my best to review it this weekend!

**Comment by fatihsen20** [comment]
I always overlook the isort rules. I am sorry 🥲

**Comment by rasbt** [comment]
No worries about that! The style checks are really not a big deal, we can address those at the very end upon merging.

**Review by rasbt — APPROVED** [review]

**Comment by fatihsen20** [comment]
Hello, will my pull request be merge? I am very excited about this 😊 @rasbt

**Comment by rasbt** [comment]
This is a fantastic PR! And sorry, will merge soon! Haven't had a chance to go over the docs yet! I have a question here: 
Screenshot 2023-03-27 at 10 44 12 AM

You mentioned

> on a large dataset.

but did you mean 

> on a small dataset.

I.e., the dataset from the example above? No worries about fixing it. I did some other tweaks in the docs and will update soon (pls don't modify the Nb in the meantime due to merge conflicts, Jupyter NB's are still a bit tricky on GitHub 😅)

**Comment by fatihsen20** [comment]
> This is a fantastic PR! And sorry, will merge soon! Haven't had a chance to go over the docs yet! I have a question here: Screenshot 2023-03-27 at 10 44 12 AM
> 
> You mentioned
> 
> > on a large dataset.
> 
> but did you mean
> 
> > on a small dataset.
> 
> I.e., the dataset from the example above? No worries about fixing it. I did some other tweaks in the docs and will update soon (pls don't modify the Nb in the meantime due to merge conflicts, Jupyter NB's are still a bit tricky on GitHub 😅)

Sorry, I probably missed it while preparing the document. Yes, there will be a small dataset in the example there. Thank you so much for your hard work 😊

**Comment by rasbt** [comment]
Oh I thought it was based on the small example at the top. So you ran the benchmark on a small hands-on dataset if I understand it correctly? I think we can leave things as is unless it's not too large and we can add it to the doc repo. What do you think?

**Comment by fatihsen20** [comment]
> Oh I thought it was based on the small example at the top. So you ran the benchmark on a small hands-on dataset if I understand it correctly? I think we can leave things as is unless it's not too large and we can add it to the doc repo. What do you think?

Yes it was based on the small example above. Since we are testing with small dataset on Doc, it may remain small. I don't think it will be a big problem if we add it this way.

**Comment by rasbt** [comment]
Ok perfect, then I'd say it's fine as is because the example is already in the notebook. 

Should be good to merge then, correct?

**Comment by fatihsen20** [comment]
> Ok perfect, then I'd say it's fine as is because the example is already in the notebook.
> 
> Should be good to merge then, correct?

Okey sir, please 😊

**Comment by rasbt** [comment]
Merged it @fatihsen20 . Thanks again for this awesome PR!
 I am hoping to make a new release version on the upcoming weekend

**Pull request merged**